app/Models/CustomField.php

```diff
@@ -180,6 +179,11 @@ public function fieldset()
     {
         return $this->belongsToMany(\App\Models\CustomFieldset::class);
     }
+   
+    public function assetModels()
+    {
+       return $this->fieldset()->with('models')->get()->pluck('models')->flatten()->unique('id'); 
```

Comment:
It would be better but I don't think there is a way we can get this eager-loadable with native Eloquent...